subprojects/language-java/src/main/java/org/gradle/api/internal/tasks/compile/incremental/recomp/ResourceChangeProcessor.java

```diff
@@ -32,6 +32,8 @@ public void processChange(InputFileDetails input, RecompilationSpec spec) {
         if (processorPath.isEmpty()) {
             return;
         }
-        spec.setFullRebuildCause("A resource changed", input.getFile());
+        if (input.isRemoved() || input.getFile().isFile()) {
```

Comment:
I would prefer using `|| !input.getFiles().isDirectory()`, since this would also cover to rebuild when a symlink is added to the sources (does that count as a resource), but I am not feeling strongly about it.